resources/analysisTools/tools/bamReadPairExtractor.pl

```diff
@@ -123,361 +124,387 @@ BEGIN
 my %count;
 my $hs = 0; #
 my $maxhs = 0;
-    
+
 while (<$bamfh>) {
-  $selected = 0;
-  #      last if ++$ctr > 10;
-  COUNT && $count{total_reads}++;
-  $fields=[(split(/\t/))];
-  $flags=get_flags($fields->[1]); 
-  next if ($flags->{NOT_PRIMARY}); #skip secondary alignments 
-  COUNT && $count{primary_alignments}++;
-
-  #      printf "Flags_paired: %s\n",$flags->{PAIRED}; 
-  #      $p = $flags->{PAIRED} ? 'paired' : 'unpaired';
-  #      print join(" ", ("1st:", $fields->[0], $fields->[1]), $p) . "\n";
-  if ($readHash->{$fields->[0]}) { # pair complete
-    $fields2 = $readHash->{$fields->[0]}[0];
-    $flags2 = $readHash->{$fields->[0]}[1];
-
-    $flags2->{'PAIRED'} || warn "First read is paired but second is not in input line $.";
-    $selected = check_selection($fields, $flags, $noiseHash); 
-    if (! $selected) {
-      $selected = check_selection($fields2,$flags2, $noiseHash);
-    }
-    if (! $selected) { 
-      delete($readHash->{$fields->[0]});
-      $hs--;
-      next;
-    } else { 
-      $matenr = (! $flags->{PAIRED}) ? 1 : ($flags->{FIRST_MATE}) ? 1 : 2; 
-      $matenr2 = (! $flags2->{PAIRED}) ? 1 : ($flags2->{FIRST_MATE}) ? 1 : 2;
-      ($matenr != $matenr2) || die "Mate number clash in input line $.";
-
-      if ((! $flags->{UNMAPPED}) && $flags->{REVERSED}) {
-        ($fields->[9] = reverse $fields->[9]) =~ tr/ACGTacgt/TGCAtgca/;
-        $fields->[10] = reverse $fields->[10];
-        if ($flags->{M_UNMAPPED}){
-          ($fields2->[9] = reverse $fields2->[9]) =~ tr/ACGTacgt/TGCAtgca/;
-          $fields2->[10] = reverse $fields2->[10];
+    $selected = 0;
+    #      last if ++$ctr > 10;
+    COUNT && $count{total_reads}++;
+    $fields = [ (split(/\t/)) ];
+    $flags = get_flags($fields->[1]);
+    next if ($flags->{NOT_PRIMARY}); #skip secondary alignments
+    COUNT && $count{primary_alignments}++;
+
+    #      printf "Flags_paired: %s\n",$flags->{PAIRED};
+    #      $p = $flags->{PAIRED} ? 'paired' : 'unpaired';
+    #      print join(" ", ("1st:", $fields->[0], $fields->[1]), $p) . "\n";
+    if ($readHash->{$fields->[0]}) {
+        # pair complete
+        $fields2 = $readHash->{$fields->[0]}[0];
+        $flags2 = $readHash->{$fields->[0]}[1];
+
+        $flags2->{'PAIRED'} || warn "First read is paired but second is not in input line $.";
+        $selected = check_selection($fields, $flags, $noiseHash);
+        if (!$selected) {
+            $selected = check_selection($fields2, $flags2, $noiseHash);
         }
-      }
-
-
-      if (! $flags2->{UNMAPPED} && $flags2->{REVERSED}) {
-        ($fields2->[9] = reverse $fields2->[9]) =~ tr/ACGTacgt/TGCAtgca/;
-        $fields2->[10] = reverse $fields2->[10];
-        if ($flags2->{M_UNMAPPED}) {
-          ($fields->[9] = reverse $fields->[9]) =~ tr/ACGTacgt/TGCAtgca/;
-          $fields->[10] = reverse $fields->[10];
+        if (!$selected) {
+            delete($readHash->{$fields->[0]});
+            $hs--;
+            next;
+        }
```

Comment:
?

Reply:
Hm, seems like the auto code cleanup in Idea is incorrectly configured for Perl.